src/traces/splom/defaults.js

```diff
@@ -0,0 +1,137 @@
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceIn, traceOut, attributes, attr, dflt);
+    }
+
+    var dimLength = handleDimensionsDefaults(traceIn, traceOut);
+    if(!dimLength) {
+        traceOut.visible = false;
+        return;
+    }
+
+    coerce('mode', traceOut._commonLength < PTS_LINESONLY ? 'lines+markers' : 'lines');
+    coerce('text');
+
+    // TODO just markers for now
+    traceOut.mode = 'markers';
+
+    if(subTypes.hasLines(traceOut)) {
+        handleLineDefaults(traceIn, traceOut, defaultColor, layout, coerce);
+        coerce('connectgaps');
+    }
+
+    if(subTypes.hasMarkers(traceOut)) {
+        handleMarkerDefaults(traceIn, traceOut, defaultColor, layout, coerce);
+
+        var isOpen = OPEN_RE.test(traceOut.marker.symbol);
+        var isBubble = subTypes.isBubble(traceOut);
+        coerce('marker.line.width', isOpen || isBubble ? 1 : 0);
+    }
+
+    // TODO not implemented yet
+    coerce('xdirection');
+    coerce('ydirection');
+
+    handleAxisDefaults(traceIn, traceOut, layout, coerce);
+
+    // TODO not implemented yet
+    coerce('showdiagonal');
+    coerce('showupperhalf');
+    coerce('showlowerhalf');
+
+    Lib.coerceSelectionMarkerOpacity(traceOut, coerce);
+};
+
+function handleDimensionsDefaults(traceIn, traceOut) {
+    var dimensionsIn = traceIn.dimensions;
+    if(!Array.isArray(dimensionsIn)) return 0;
+
+    var dimLength = dimensionsIn.length;
+    var commonLength = 0;
+    var dimensionsOut = traceOut.dimensions = new Array(dimLength);
+    var dimIn;
+    var dimOut;
+    var i;
+
+    function coerce(attr, dflt) {
+        return Lib.coerce(dimIn, dimOut, attributes.dimensions, attr, dflt);
+    }
+
+    for(i = 0; i < dimLength; i++) {
+        dimIn = dimensionsIn[i];
+        dimOut = dimensionsOut[i] = {};
+
+        var visible = coerce('visible');
+        if(!visible) continue;
+
+        var values = coerce('values');
+        if(!values || !values.length) {
+            dimOut.visible = false;
+            continue;
+        }
+
+        coerce('label');
+
+        commonLength = Math.max(commonLength, values.length);
+        dimOut._index = i;
+    }
+
+    for(i = 0; i < dimLength; i++) {
+        dimOut = dimensionsOut[i];
+        if(dimOut.visible) dimOut._length = commonLength;
+    }
+
+    traceOut._commonLength = commonLength;
+
+    return dimensionsOut.length;
+}
+
+function handleAxisDefaults(traceIn, traceOut, layout, coerce) {
+    var dimensions = traceOut.dimensions;
+    var dimLength = dimensions.length;
+    var xaxesDflt = new Array(dimLength);
+    var yaxesDflt = new Array(dimLength);
+    var i;
+
+    for(i = 0; i < dimLength; i++) {
+        xaxesDflt[i] = 'x' + (i ? i + 1 : '');
+        yaxesDflt[i] = 'y' + (i ? i + 1 : '');
+    }
+
+    var xaxes = coerce('xaxes', xaxesDflt);
+    var yaxes = coerce('yaxes', yaxesDflt);
+
+    // TODO what to do when xaxes.length or yaxes.length !== dimLength ???
```

Comment:
I guess if `dimLength` is longer it's easy to just drop the later dims. What if `xaxes.length !== yaxes.length` or `dimLength` is shorter than `xaxes` or `yaxes`? The *easy* thing to do is just truncate them all to the same length, and perhaps until we have a clear use case otherwise we should just do that.

Related though: if you have several `splom` traces but they don't share all the same dimensions, you could of course handle this by specifying `xaxes` and `yaxes` for each, but could you also just pass in an empty dimension for ones you want to skip? If that takes more than 5 minutes to implement though, skip it for now and leave it as an item for future expansion.

Reply:
Oh you already have `dimensions[i].visible: false`, nice! That should take care of my "related" point, it'll still reserve x/y axes for these dimensions, right?

Reply:
> The easy thing to do is just truncate them all to the same length, and perhaps until we have a clear use case otherwise we should just do that.

That sounds right to me. I'll go with that unless anyone disagrees.

> That should take care of my "related" point, it'll still reserve x/y axes for these dimensions, right?

Yes that _should_ take care of it, but I haven't tried that yet. Putting that on my TODO list :memo:

Reply:
.... thanks for all these comments by the way. I hope you don't the `_hasOnlyLargeSplom` flag too hacky.

Reply:
done in 07d59d553bc4bd5c762a72a6e8e34b0f489d0e80